File: templates/fsharp/app-mvvm/Views/MainWindow.axaml.fs
namespace AvaloniaAppTemplate.Views

open Avalonia
open Avalonia.Controls
open Avalonia.Markup.Xaml

type MainWindow () as self = 
    inherit Window ()
//-:cnd:noEmit
#if DEBUG
    do self.AttachDevTools()
#endif
//+:cnd:noEmit
    do AvaloniaXamlLoader.Load self

File: templates/csharp/app-mvvm/Views/MainWindow.axaml.cs
using Avalonia;
using Avalonia.Controls;
using Avalonia.Markup.Xaml;

namespace AvaloniaAppTemplate.Views
{
    public partial class MainWindow : Window
    {
        public MainWindow()
        {
            InitializeComponent();
//-:cnd:noEmit
#if DEBUG
            this.AttachDevTools();
#endif
//+:cnd:noEmit
        }

        private void InitializeComponent()
        {
            AvaloniaXamlLoader.Load(this);
        }
    }
}